docs/rollup-boost-ha.md

```diff
@@ -0,0 +1,328 @@
+
+Note that `rollup-boost` does not currently feature a block selection policy and will optimistically select the builder's block for validation. In the event of a bug in the builder, it is possible valid but undesirable blocks (eg. empty blocks) are produced. Without a block selection policy, `rollup-boost` will prefer these blocks over the default execution client. Proper monitoring alerting can help mitigate this but further designs should be explored to introduce safeguards into `rollup-boost` directly rather than relying on the builder implementation being correct.
+
+Below is a happy path sequence diagram illustrating how `rollup-boost` facilitates payload construction/validation:
+
+```mermaid
+sequenceDiagram
+    participant CL as op-node
+    participant RB as rollup-boost
+    participant EL as op-geth
+    participant B as Builder
+
+    %% FCU forwarded with payload attrs
+    Note over CL: FCU with Attributes
+    CL->>RB: engine_forkChoiceUpdated(..., Attrs)
+    RB->>EL: engine_forkChoiceUpdated(..., Attrs)
+    RB->>B: engine_forkChoiceUpdated(..., Attrs)
+    B->>B: Start payload job
+
+    %% get_payload from builders
+    CL->>RB: engine_getPayload(PayloadId)
+    RB->>EL: engine_getPayload(PayloadId)
+    RB->>B: engine_getPayload(PayloadId)
+    B-->>RB: {executionPayload, blockValue}
+    EL-->>RB: {executionPayload, blockValue}
+
+    %% block selection policy
+    RB->>RB: Optimistically select builder block 
+
+    %% Validate best payload
+    Note over RB, EL: Validate builder block
+    RB->>EL: engine_newPayload(ExecutionPayload)
+    EL->>RB: {status: VALID, ...}
+
+    %% Propagate block to builders
+    Note over RB: Propagate new block to builder
+    RB->>B: engine_newPayload(ExecutionPayload)
+
+    %% Return final payload to op-node
+    RB->>CL: {executionPayload, blockValue}
+
+```
+
+### Failure Scenarios
+
+<details>
+<summary> Failure Scenarios</summary>
+
+Below is a high level summary of how each failure scenario is handled. All existing failure modes assumed by upstream `op-conductor` are maintained:
+
+| Failure Scenario | Category | Scenario and Solution |
+| --- | --- | --- |
+| Leader Sequencer Fails | Sequencer Failure | Conductor will detect sequencer failure and start to transfer leadership to another node, which will start sequencing instead. Currently,`op-conductor` will not know if the newly elected sequencer has a healthy builder. In the event that `op-conductor` selects a sequencer with an unhealthy builder, manual intervention is needed to either switch the leader or recover the builder. Once the originally failed sequencer is back online, it will join the sequencer set as a follower. |
```

Comment:
depending on if we ask conductor to monitor builder health or not, if we did as described above, then op-conductor will know that the newly elected leader is not healthy and then continue to transfer leader until it finds a healthy one (if it was able to).

In the case that we don't ask conductor to monitor builder health, yeah, it'll treat the new sequencer as healthy if its local builder is healthy. Which is the point (conductor cannot act on information it does not know).

So if we want to avoid manual intervention here, we'll need to add builder health check logic to conductor

Reply:
> So if we want to avoid manual intervention here, we'll need to add builder health check logic to conductor
Yes I think we should avoid manual intervention here. The current plan is to [add `rollup-boost` monitoring to `op-conductor`](https://github.com/ethereum-optimism/optimism/pull/15316). With #142 merged, this is now unblocked.